src/http-api/resources/object.js

```diff
@@ -30,41 +30,18 @@ exports.parseKey = (request, reply) => {
   }
 }
 
-exports.new = {
-  // pre request handler that parses the args and returns `template` which is assigned to `request.pre.args`
-  parseArgs: (request, reply) => {
-    // TODO improve this validation once request.server.app.ipfs.object.new supports templates
-    if (request.query.arg === '') {
+exports.new = (request, reply) => {
+  request.server.app.ipfs.object.new((err, node) => {
+    if (err) {
+      log.error(err)
       return reply({
-        Message: `template \'${request.query.arg}\' not found`,
+        Message: 'Failed to create object: ' + err,
```

Comment:
You changed a template string to a string concat? I'll need @dignifiedquire private key signature in order to accept that this PR came indeed by the real @dignifiedquire :D